src/trackers/classes/commonRequest.js

```diff
@@ -106,6 +136,25 @@ function _finalize (request, totalSites) {
 
     request.exampleSites = getExampleSites([...request.pages], sharedData.config.includeExampleSites)
     delete request.pages
+
+    Object.values(request.firstPartyCookies).forEach(cookie => {
+        cookie.length = Math.round(cookie.lengthSum / cookie.pages)
+        cookie.ttl = cookie.ttl.sort()[Math.floor(cookie.ttl.length / 2)]
```

Comment:
I don't think it's a big deal, but maybe it makes sense to use a "full version" of a median instead (although that introduces this NaN + Number edge-case that I mentioned):

```js
function median(values) {
    if (values.length === 0) return 0;

    values.sort((a, b) => a - b);

    const half = Math.floor(values.length / 2);

    if (values.length % 2)
        return values[half];

    return (values[half - 1] + values[half]) / 2.0;
}
```

Reply:
It seems we already use `mathjs` which includes a `median` implementation which is likely robust. We could use that: https://github.com/duckduckgo/tracker-radar-detector/blob/main/src/trackers/classes/crawl.js#L2

Reply:
Yeah, even better 👍

Reply:
I quickly tested the mathjs median function, and it seems it will always return `NaN` if 1 or more `NaN` elements exists in the list. My implementation still works for this case.